tests/test_ir.py

```diff
@@ -997,3 +997,23 @@ def test_replace_by_if():
 
     assert set(u.operation for u in a.uses) == {c}
     assert set(u.operation for u in d.uses) == {b}
+
+
+def test_find_ancestor_op_in_block_has_ancestor():
+    op1 = test.TestOp()
+    op2 = test.TestOp(regions=[Region(Block([op1]))])
+
+    blk_top = Block([op2])
+    test.TestOp(regions=[Region(blk_top)])
+
+    assert blk_top.find_ancestor_op_in_block(op1) is op2
+
+
+def test_find_ancestor_op_in_block_has_no_ancestor():
+    op1 = test.TestOp()
+    op2 = test.TestOp(regions=[Region(Block([op1]))])
+
+    blk_top = Block([op2])
+    op3 = test.TestOp(regions=[Region(blk_top)])
+
+    assert blk_top.find_ancestor_op_in_block(op3) is None
```

Comment:
I just realized these can be folded. Fewer functions is better for this sort of thing as each one has a bit of overhead when running the tests, which adds up over time.
```suggestion
def test_find_ancestor_op_in_block():
    op1 = test.TestOp()
    op2 = test.TestOp(regions=[Region(Block([op1]))])

    blk_top = Block([op2])
    op3 = test.TestOp(regions=[Region(blk_top)])

    assert blk_top.find_ancestor_op_in_block(op1) is op2
    assert blk_top.find_ancestor_op_in_block(op3) is None
```

Reply:
I think it would be nice to see a more nested scenario. If op3 was in a block `b` then my understanding is that `b.find_ancestor_op_in_block(op1)` should return `op3` and not `op2`

Reply:
That is correct. I can add another level of nesting, no problem